junebug/stores.py

```diff
@@ -142,3 +151,46 @@ def get_statuses(self, channel_id):
             (k, TransportStatus.from_json(v))
             for k, v in statuses.iteritems()
         ))
+
+
+class MessageRateStore(BaseStore):
+    '''Gets called everytime a message should be counted, and can return the
+    current messages per second.'''
+
+    def get_seconds(self):
+        return time.time()
+
+    def get_key(self, channel_id, label, bucket):
+        return super(MessageRateStore, self).get_key(
+            channel_id, label, str(bucket))
+
+    def _get_current_key(self, channel_id, label, bucket_size):
+        bucket = int(self.get_seconds() / bucket_size)
+        return self.get_key(channel_id, label, bucket)
+
+    def _get_last_key(self, channel_id, label, bucket_size):
+        bucket = int(self.get_seconds() / bucket_size) - 1
+        return self.get_key(channel_id, label, bucket)
+
+    def increment(self, channel_id, label, bucket_size):
+        '''Increments the correct counter. Should be called whenever a message
+        that should be counted is received.
+
+        Note: bucket_size should be kept constant for each channel_id and label
+        combination. Changing bucket sizes results in undefined behaviour.'''
+        key = self._get_current_key(channel_id, label, bucket_size)
+        self.ttl = bucket_size * 2
+        return self.increment_id(key)
+
+    @inlineCallbacks
+    def get_messages_per_second(self, channel_id, label, bucket_size):
+        '''Gets the current message rate in messages per second.
+
+        Note: bucket_size should be kept constant for each channel_id and label
+        combination. Changing bucket sizes results in undefined behaviour.'''
+        key = self._get_last_key(channel_id, label, bucket_size)
+        self.ttl = None
```

Comment:
I feel that juggling `self.ttl` here is not really worth it. If a few extra redis metric keys live for a bit longer, it's not a much of an issue, but if all metric keys don't have expirey set because we forget to set the TTL back to a value, that's a problem?